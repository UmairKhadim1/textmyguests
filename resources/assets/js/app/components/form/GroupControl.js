import React, {useEffect} from 'react';
import { Checkbox } from 'antd';
import './GroupControl.css';


const GroupControl = ({
  groups = [],
  selected = {},
  showAll = true,
  disableIfAll = false,
  is_testGroup = false,
  is_hide_testGroup = false,
  onChange,
  onClick,
}: {
  groups: Array<Object>,
  selected: Object,
  showAll: boolean,
  disableIfAll: boolean,
  is_testGroup:boolean,
  is_hide_testGroup: boolean,
  onChange: Function,
  onClick: Function,
}) => {
  const handleAllGuests = e => {
    const add = e.target.checked;
    let newSelected = {};
    if (add) {
      newSelected = { all: true };
    }
    onChange(newSelected);
    onClick(newSelected)//this will send the selected group
  };

  // useEffect(()=> {
  //   if(is_testGroup){//if the test group only is enabled then only test group will be checked
  //     groups.map(group => {
  //       if(group.is_testGroup) {
  //         const data = {}
  //         const id = group.id
  //         let gid = id.toString()
  //         data[gid] = true
  //         onChange(data)
  //       }
  //     })
  //   }
  //   else {
  //     onChange([])
  //   }
  // }, [is_testGroup])

  const handleGroupSelect = (id, e) => {
    const add = e.target.checked;
    // Copy the selected map
    
      const newSelected = { ...selected };
      // Clean state
      delete newSelected.all; // pollute the condition
      if (add) {
        // We save the selected state
        newSelected[id.toString()] = true;
      } else {
        // We just remove this one
        delete newSelected[id.toString()];
      }
      onChange(newSelected);
      onClick(newSelected) //this will send the selected group
  };

  const eventGroups = groups.filter(group => !group.is_all);
  const all = !!selected.all;
  return (
    <div>
      {showAll ? (
        <div key="all">
          <Checkbox checked={all} onChange={handleAllGuests} >
            All Guests
          </Checkbox>
        </div>
      ) : null}
      {eventGroups.map(group => (
        <div key={group.id} className={is_hide_testGroup && group.is_testGroup ? 'd-none': 'd-block'}>
          <Checkbox
            disabled= {disableIfAll && all }
            checked={ !!selected[group.id.toString()]}
            onChange={e => handleGroupSelect(group.id, e, group.name)}>
            {group.name}
          </Checkbox>
        </div>
      ))}
    </div>
  );
};

export default GroupControl;
