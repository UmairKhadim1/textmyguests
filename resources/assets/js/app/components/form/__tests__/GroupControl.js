import React from 'react';
import { shallow } from 'enzyme';
import GroupControl from '../GroupControl';

describe('GroupControl Tests', () => {
  it('should render without crash', () => {
    const groups = [
      {
        id: 1,
        name: 'Group 1',
      },
    ];
    const wrapper = shallow(<GroupControl groups={groups} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without the all guest checkbox', () => {
    const groups = [
      {
        id: 1,
        name: 'Group 1',
      },
    ];
    const wrapper = shallow(<GroupControl groups={groups} showAll={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should select only one group', () => {
    const groups = [
      {
        id: 1,
        name: 'Group 1',
      },
      {
        id: 2,
        name: 'Group 2',
      },
    ];
    const handler = jest.fn();
    const wrapper = shallow(
      <GroupControl groups={groups} onChange={handler} showAll={false} />
    );

  });
});
