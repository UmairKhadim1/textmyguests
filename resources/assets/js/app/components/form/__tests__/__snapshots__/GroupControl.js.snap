// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupControl Tests should render without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupControl
    disableIfAll={false}
    groups={
      Array [
        Object {
          "id": 1,
          "name": "Group",
        },
      ]
    }
    onChange={[Function]}
    selected={Object {}}
    showAll={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Checkbox
            checked={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            All Guests
          </Checkbox>
        </div>,
        Array [
          <div>
            <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>
          </div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "all",
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            All Guests
          </Checkbox>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": "All Guests",
            "indeterminate": false,
            "onChange": [Function],
            "prefixCls": "ant-checkbox",
          },
          "ref": null,
          "rendered": "All Guests",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            disabled={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            Group
          </Checkbox>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": "Group",
            "disabled": false,
            "indeterminate": false,
            "onChange": [Function],
            "prefixCls": "ant-checkbox",
          },
          "ref": null,
          "rendered": "Group",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Checkbox
              checked={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              All Guests
            </Checkbox>
          </div>,
          Array [
            <div>
              <Checkbox
                checked={false}
                disabled={false}
                indeterminate={false}
                onChange={[Function]}
                prefixCls="ant-checkbox"
              >
                Group
              </Checkbox>
            </div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "all",
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              All Guests
            </Checkbox>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "children": "All Guests",
              "indeterminate": false,
              "onChange": [Function],
              "prefixCls": "ant-checkbox",
            },
            "ref": null,
            "rendered": "All Guests",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "children": "Group",
              "disabled": false,
              "indeterminate": false,
              "onChange": [Function],
              "prefixCls": "ant-checkbox",
            },
            "ref": null,
            "rendered": "Group",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`GroupControl Tests should render without scratch 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupControl
    disableIfAll={false}
    groups={
      Array [
        Object {
          "id": 1,
          "name": "Group",
        },
      ]
    }
    onChange={[Function]}
    selected={Object {}}
    showAll={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Checkbox
            checked={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            All Guests
          </Checkbox>
        </div>,
        Array [
          <div>
            <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>
          </div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "all",
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            All Guests
          </Checkbox>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": "All Guests",
            "indeterminate": false,
            "onChange": [Function],
            "prefixCls": "ant-checkbox",
          },
          "ref": null,
          "rendered": "All Guests",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            disabled={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            Group
          </Checkbox>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": "Group",
            "disabled": false,
            "indeterminate": false,
            "onChange": [Function],
            "prefixCls": "ant-checkbox",
          },
          "ref": null,
          "rendered": "Group",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Checkbox
              checked={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              All Guests
            </Checkbox>
          </div>,
          Array [
            <div>
              <Checkbox
                checked={false}
                disabled={false}
                indeterminate={false}
                onChange={[Function]}
                prefixCls="ant-checkbox"
              >
                Group
              </Checkbox>
            </div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "all",
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              All Guests
            </Checkbox>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "children": "All Guests",
              "indeterminate": false,
              "onChange": [Function],
              "prefixCls": "ant-checkbox",
            },
            "ref": null,
            "rendered": "All Guests",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "children": "Group",
              "disabled": false,
              "indeterminate": false,
              "onChange": [Function],
              "prefixCls": "ant-checkbox",
            },
            "ref": null,
            "rendered": "Group",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`GroupControl Tests should render without the all guest checkbox 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupControl
    disableIfAll={false}
    groups={
      Array [
        Object {
          "id": 1,
          "name": "Group",
        },
      ]
    }
    onChange={[Function]}
    selected={Object {}}
    showAll={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        Array [
          <div>
            <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>
          </div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Checkbox
            checked={false}
            disabled={false}
            indeterminate={false}
            onChange={[Function]}
            prefixCls="ant-checkbox"
          >
            Group
          </Checkbox>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": "Group",
            "disabled": false,
            "indeterminate": false,
            "onChange": [Function],
            "prefixCls": "ant-checkbox",
          },
          "ref": null,
          "rendered": "Group",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          Array [
            <div>
              <Checkbox
                checked={false}
                disabled={false}
                indeterminate={false}
                onChange={[Function]}
                prefixCls="ant-checkbox"
              >
                Group
              </Checkbox>
            </div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Checkbox
              checked={false}
              disabled={false}
              indeterminate={false}
              onChange={[Function]}
              prefixCls="ant-checkbox"
            >
              Group
            </Checkbox>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "children": "Group",
              "disabled": false,
              "indeterminate": false,
              "onChange": [Function],
              "prefixCls": "ant-checkbox",
            },
            "ref": null,
            "rendered": "Group",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
